syntax = "proto3";

package greeter;

// Option specifying the Go package path where the generated files will reside.

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply);
  rpc GetUsers(GetUsersRequest) returns (GetUsersResponse) {}
  rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse){}
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse){}
  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse){}
  rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse){}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greeting.
message HelloReply {
  string message = 1;
}


message User{
  string id = 1;
  string name = 2;
  string email = 3;
  string password = 4;
  string phone = 5;
  string address = 6;
  string role = 7;
  string status =8;
  string created_at = 9;
  string updated_at = 10;
}

message GetUsersRequest {
}

message GetUsersResponse {
  repeated User users = 1;
}

message GetUserByIdRequest {
  string id = 1;
}

message GetUserByIdResponse {
  User user = 1;
}

message CreateUserRequest {
  User user = 1;
}

message CreateUserResponse {
  User user = 1;
}

message UpdateUserRequest {
  User user = 1;
}

message UpdateUserResponse {
  User user = 1;
}

message DeleteUserResponse {
  User user = 1;
}

message DeleteUserRequest {
  string id = 1;
}

